### ISSUE TITLE
How to normalize the 2d coordinate system?

### ISSUE THREAD

**Issue by h0ngxuanli** [nitinkamra1992/FQA#1]
Hi, Nitin Kamra!

I find that you denormalize the pred with a scale number for the whole dataset.  I am a little confused about the way you normalizing the coordinate data, i.e. [[x1, y1], [x2, y2], ...] 🤔.

I am wondering if you could share the data preprocessing code？

**Comment by nitinkamra1992** [comment]
The dataset pre-processing codes are a bit more scattered and specific to each dataset's format. So we did not put them in the repo to keep the uploaded code clean. However, I can explain what our scale normalization does. The trajectory datasets have a variable range of x and y coordinates. We have normalized each dataset separately by scaling them such that all x and y coordinates input to the neural nets are always b/w a reasonable range (e.g. [-1,1]). We used the same scaling constant for both x and y axis for a dataset to ensure rotational symmetry in the data. This means that the next positions predicted by neural nets also come with the same scale normalization. Before we calculate and display the metrics, we scale up the neural net predictions and the ground truth coordinates back by the same normalizing factor to ensure that they are on the scale of the actual dataset.
We have provided the normalized and processed datasets along with the normalization constant for each of them in the repo. The constant helps de-normalize the predictions etc. before displaying metrics.
